### PR TITLE
Add optimization in GetHashCode for Int16 and SByte

### PR DESCRIPTION
**Pull request by ichensky**
As discussed in https://github.com/dotnet/corefx/pull/34123

### CODE REVIEW

**Comment by danmosemsft** [comment]
Notwithstanding [this comment](https://github.com/dotnet/corefx/pull/34123#issuecomment-448094207). I guess anyone doing that in .NET would have to use the least significant byte.

File: src/System.Private.CoreLib/shared/System/SByte.cs [hunk 1]
```diff
@@ -66,7 +66,7 @@ public bool Equals(sbyte obj)
         // Gets a hash code for this instance.
         public override int GetHashCode()
         {
-            return ((int)m_value ^ (int)m_value << 8);
+            return (int)m_value;
```
Comment (tannergooding) [inline src/System.Private.CoreLib/shared/System/SByte.cs]: nit: Would be nice to remove the unnecessary cast to `(int)` since C# will do it implicitly.

Comment (jkotas) [reply src/System.Private.CoreLib/shared/System/SByte.cs]: I have pushed the fix for this.

**Comment by danmosemsft** [comment]
Note to self - if we did want to spread the hashcode over more bits, we have `HashCode.Combine<T1>(T1 value)` for this purpose already:
```c#
        public static int Combine<T1>(T1 value1)
        {
            // Provide a way of diffusing bits from something with a limited
            // input hash space. For example, many enums only have a few
            // possible hashes, only using the bottom few bits of the code. Some
            // collections are built on the assumption that hashes are spread
            // over a larger space, so diffusing the bits may help the
            // collection work more efficiently.
```

**Pull request merged**